app/shared/app-data/src/desktopMain/kotlin/data/source/media/resolver/DesktopWebVideoSourceResolver.kt

```diff
@@ -120,223 +135,224 @@ class DesktopWebVideoSourceResolver(
+                        runOnEventDispatcherThread { browser.loadURL(url) }
+                        return false
                     }
                 }
+            }
 
-                driver.get(pageUrl)
+            runOnEventDispatcherThread {
+                // 获取到 DevTools 后就可以移除 load handler 了
+                browser.client.removeLoadHandler()
+                
+                // set cookie
+                val cookieManager = CefCookieManager.getGlobalManager()
+                val url = Url(pageUrl)
+                for (cookie in config.cookies) {
+                    val ktorCookie = parseServerSetCookieHeader(cookie)
+                    cookieManager.setCookie(url.host, ktorCookie.toCefCookie())
+                }
 
-                for (t in webViewConfig.cookies) {
-                    try {
-                        val url = Url(pageUrl)
-                        driver.manage().addCookie(
-                            parseServerSetCookieHeader(t)
-                                .toDriverCookie(domain = url.host),
-                        )
-                    } catch (e: Exception) {
-                        logger.error(e) { "Failed to parse or add cookie, see cause" }
+                // setup network
+                devTools.executeDevToolsMethod("Network.enable")
+                devTools.addEventListener { name, message ->
+                    if (name != "Network.requestWillBeSent") return@addEventListener
+                    val data = json.decodeFromString(RequestWillBeSent.serializer(), message)
+
+                    if (handleUrl(data.request.url)) {
+                        logger.info { "Found video resource via DevTools: ${data.request.url}" }
                     }
                 }
-            }
+                devTools.executeDevToolsMethod("Network.clearBrowserCache")
 
+                logger.info { "Fetching $pageUrl" }
+                // load real page after setting up DevTools
+                browser.loadURL(pageUrl)
+            }
+            
             deferred.await()
         } catch (e: Throwable) {
+            logger.error(e) { "Failed to get video url." }
             if (deferred.isActive) {
-                deferred.cancel() // will quit driver
+                deferred.cancel()
             }
-            throw e
-        }
-    }
-
-    private fun io.ktor.http.Cookie.toDriverCookie(domain: String): Cookie {
-        return Cookie(name, value, domain, path, expires?.toJvmDate(), secure, httpOnly)
-    }
-
-    private fun createDriver(): RemoteWebDriver {
-        return getPreferredDriverFactory()
-            .runCatching {
-                create(videoResolverSettings, proxyConfig)
+            null
+        } finally {
+            // close browser and client asynchronously.
+            runOnEventDispatcherThread {
+                browser.close(true)
+                client.dispose()
+                logger.info { "CEF client is disposed." }
```

Comment:
Possibly log in main (time-critical) thread? Logging can be extremely slow since you'll write onto disk